File: 1-twoSum/Cpp/twoSum_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "twoSum.cpp"


TEST_CASE("testing the twoSum function") {
    std::vector<int> inArray = {2,7,12,4,-12,-13};
    std::vector<int> inTargets = {9,6,19,11,0,-25};
    std::vector<std::vector<int>> wants = {{0,1}, {0,3}, {1,2}, {1,3}, {2,4}, {4,5}};

    for(int i =0; i < inTargets.size(); i++){
        auto got = Solution().twoSum(inArray, inTargets[i]);
        
        CHECK(got == wants[i]);
    };
}

    

File: 1-twoSum/Python3/twoSum_test.py
'''testing'''

import unittest
from twoSum import Solution

class TestTwoSum(unittest.TestCase):

    def test_zero_case(self):
        data = [0, 0]
        target = 0
        want = [0,1]
        got = Solution().twoSum(data, target)
        self.assertEqual(got, want)

    def test_multi_case(self):
        data = [2,7,12,4]
        targets = [9, 6, 19, 16]
        wants = [[0,1], [0,3], [1,2], [2,3]]
        for (target, want) in zip(targets, wants):
            got = Solution().twoSum(data, target)
            self.assertEqual(got, want)

if __name__ == '__main__':
    unittest.main()